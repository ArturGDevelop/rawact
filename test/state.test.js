import React from "react";
import ReactDOM from "react-dom";
import * as State from './cases/state';
import renderHelper from "./renderHelper";

describe('state', () => {
	it('renders correctly each time state changes', async () => {
		const { rendered } = renderHelper(<State.Hello name="world"/>);

		expect(rendered).toMatchSnapshot();

		State.runTick();
		await Promise.resolve();

		expect(rendered).toMatchSnapshot();

		State.runTick();
		await Promise.resolve();

		expect(rendered).toMatchSnapshot();
	});
});
