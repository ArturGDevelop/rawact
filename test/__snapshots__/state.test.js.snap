// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`state renders correctly each time state changes 1`] = `
<div>
  <p>
    Hello 
    world
    ! Called x
    1
  </p>
</div>
`;

exports[`state renders correctly each time state changes 2`] = `
<div>
  <p>
    Hello 
    world
    ! Called x
    2
  </p>
</div>
`;

exports[`state renders correctly each time state changes 3`] = `
<div>
  <p>
    Hello 
    world
    ! Called x
    3
  </p>
</div>
`;
